locked-nft/contracts/NFTLocker.cdc

```diff
@@ -89,27 +93,177 @@ access(all) contract NFTLocker {
         return false
     }
 
-    /// The path to the NFTLocker Admin resource belonging to the Account
-    /// which the contract is deployed on
-    access(all) view fun GetAdminStoragePath(): StoragePath {
+    /// The path to the Admin resource belonging to the account where this contract is deployed
+    ///
+    access(all) view fun getAdminStoragePath(): StoragePath {
         return /storage/NFTLockerAdmin
     }
 
+    /// The path to the ReceiverCollector resource belonging to the account where this contract is deployed
+    ///
+    access(all) view fun getReceiverCollectorStoragePath(): StoragePath {
+        return /storage/NFTLockerAdminReceiverCollector
+    }
+
+    /// Return an unauthorized reference to the admin's ReceiverCollector resource if it exists
+    ///
+    access(all) view fun borrowAdminReceiverCollectorPublic(): &ReceiverCollector? {
+        return self.account.storage.borrow<&ReceiverCollector>(from: NFTLocker.getReceiverCollectorStoragePath())
+    }
+
+    /// Interface for depositing NFTs to authorized receivers
+    ///
+    access(all) struct interface IAuthorizedDepositHandler {
```

Comment:
New struct interface to ensure a valid deposit method to be used for the receiver in `unlockWithAuthorizedDeposit` is defined